packages/payment/src/migrations/Migration20240225134525.ts

```diff
@@ -0,0 +1,162 @@
+import { generatePostgresAlterColummnIfExistStatement } from "@medusajs/utils"
+import { Migration } from "@mikro-orm/migrations"
+
+export class Migration20240225134525 extends Migration {
+  async up(): Promise<void> {
+    this.addSql(
+      `
+CREATE TABLE IF NOT EXISTS "payment_collection" (
```

Comment:
Let's try to break this apart in multiple `addSql` statements to aid readability.

One approach we can take here is to separate a clean v2 install, from the one that migrates from v1 to v2, see [this PR](https://github.com/medusajs/medusa/pull/6515/files#diff-e20b7a5cf6444556be24a5c3985f01388e9e4e3ecdc12f1001503657f871c57f).

The benefit of this is two-fold:
1. It makes the code more readable
2. We can remove the v1-to-v2 migration code later on when we fully deprecate v1.

Reply:
👍

Reply:
We will need to go through all migrations before releasing v2 to ensure everything is correct and they are clean and easy to read. Therefore, I would like to punt all work related to migrations so that we do not do double the work. 

The idea of getting v1 tables up to speed with v2 schemas was introduced after the product and pricing module, so I know we already have some work to do in this area.

Reply:
@olivermrbl yeah I think we'll need to revisit every single migration and make sure we are not missing anything, but it's good to at least have a solid initial setup in place, but of course feel free to do as much as you think it's necessary :)